Source: Written by Alex Elsayed for Exherbo
Upstream: No bug tracker, emailed to maintainer
Reason: Work around testapp failing to find -lQtCore
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 710791b..9d3a044 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,10 @@ install(FILES ${CMAKE_BINARY_DIR}/dbusmenu-qt.pc
     DESTINATION lib${LIB_SUFFIX}/pkgconfig
     )
 
+option(BUILD_TESTS "Build tests" ON)
+
 add_subdirectory(src)
-add_subdirectory(tests)
+if (BUILD_TESTS)
+	add_subdirectory(tests)
+endif (BUILD_TESTS)
 add_subdirectory(tools)
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index c150cc7..3322657 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,7 +1,11 @@
-find_package(QJSON)
-if (QJSON_FOUND)
-    message(STATUS "QJSON found, testapp will be built")
-    add_subdirectory(testapp)
-else (QJSON_FOUND)
-    message(STATUS "QJSON not found, testapp will not be built")
-endif (QJSON_FOUND)
+if (BUILD_TESTS)
+	find_package(QJSON)
+	if (QJSON_FOUND)
+		message(STATUS "QJSON found and tests enabled, testapp will be built")
+		add_subdirectory(testapp)
+	else (QJSON_FOUND)
+		message(STATUS "QJSON not found, testapp will not be built")
+	endif (QJSON_FOUND)
+else (BUILD_TESTS)
+	message(STATUS "Tests disabled, testapp will not be built")
+endif (BUILD_TESTS)
